daemon/logger/fluentd/fluentd_test.go

```diff
@@ -199,3 +206,127 @@ func TestValidateLogOptAddress(t *testing.T) {
+	defer l.Close()
+
+	// This is to guard against potential run-away test scenario so that a future change
+	// doesn't cause the tests suite to timeout. It "fluentd-write-timeout" is not set, this
+	// test will be blocked on the socket write operation. The timeout guards against that.
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	defer cancel()
+
+	var connectedWG sync.WaitGroup
+	connectedWG.Add(1)
+
+	// Start accepting connections.
+	go func(ctx context.Context, wg *sync.WaitGroup) {
+		wg.Done()
+
+		for {
+			conn, err := l.Accept()
+			if err != nil {
+				// Unable to process the connection. This can happen if the connection is closed.
+				select {
+				case <-ctx.Done():
+					// If the context is canceled, there's nothing for us to do here.
+					return
+				default:
+					t.Logf("Unable to accept connection: %v", err)
+					continue
+				}
+			}
+
+			// Handle an incoming connection. We're essentially blackholing this by not reading from
+			// or writing to the connection.
+			go func(ctx context.Context, conn net.Conn) {
+				// Simulate unresponsive server: do nothing with the connection
+				<-ctx.Done()
+				_ = conn.Close()
+			}(ctx, conn)
+		}
+	}(ctx, &connectedWG)
+
+	f, err := New(logger.Info{
+		ContainerName: "/test-container",
+		ContainerID:   "container-abcdefghijklmnopqrstuvwxyz01234567890",
+		Config: map[string]string{
+			"fluentd-address": "unix://" + socketFile,
+			"tag":             "{{.Name}}/{{.FullID}}",
+			// Disabling async behavior with limited retries and buffer size lets
+			// us test this in a more preditable manner for failures. Otherwise,
+			// write errors could be silently consumed. The "fluentd-write-timeout"
+			// flag should be equally effective regardless of async being enabled/disabled.
+			"fluentd-async":         "false",
+			"fluentd-max-retries":   "1",
+			"fluentd-retry-wait":    "10ms",
+			"fluentd-buffer-limit":  "1",
+			"fluentd-write-timeout": "1ms",
+		},
+	})
+	assert.NilError(t, err)
+	defer f.Close()
+
+	// Ensure that the server is ready to accept connections since we have disabled async mode
+	// in fluentd options.
+	connectedWG.Wait()
+
+	// Attempt writing 1MiB worth of log data (all 0's) repeatedly. We should see a failure
+	// after the 1st or the 2nd attempt depending on when the connection's write buffer gets
+	// filled up.
+	// If we don't set a write timeout on the connection, this will hang forever. But, because
+	// we have a write timeout, we expect the `Log` method to return an error.
+	data := make([]byte, 1024*1024)
+	for range 10 {
+		err = f.Log(&logger.Message{
+			Line:      data,
+			Timestamp: time.Now(),
+		})
+		if err != nil {
+			break
+		}
+	}
+
+	assert.Check(t, err != nil)
+	// Check if error string matches the pattern "fluent#write: failed to write after %d attempts"
+	// where %d is a number.
+	assert.Check(t, cmp.Regexp(`fluent#write: failed to write after \d+ attempts`, err.Error()))
```

Comment:
It's fine to simplify to just "error contains"; checking that prefix should be "good enough" to know we have the expected error;

```go
assert.ErrorContains(t, err, "fluent#write: failed to write after")
```

When doing so, you can also remove the `assert.Check(t, err != nil)` above (which is then redundant)

Reply:
Yeah, seems like `ErrorContains` does a `err != nil` check as well -> https://github.com/gotestyourself/gotest.tools/blob/0b81523ff268a1f1b0baf4a5da67e42fbb86880b/assert/cmp/compare.go#L233

Will modify this.